### PR TITLE
Soft 123 add a global store instead of localstorage & make cart & user auth cart

### PR DESCRIPTION
**Pull request by mohamedahmed-cloud**

### CODE REVIEW

File: components/Navbar/Navbar.tsx [hunk 1]
```diff
@@ -9,6 +9,7 @@ import LanguageAndTheme from './LanguageAndTheme';
 import ResonsiveIcon from './ResponsiveIcon';
 import BrandNameAndLogo from './Brand';
 import HoverDropDown from './HoverDropDown';
+import Card from './Cart';
```
Comment (MuhammadSawalhy) [inline components/Navbar/Navbar.tsx:13]: Typo again!

File: navigation.ts [hunk 2]
```diff
@@ -1,27 +1,10 @@
...
-  '/tmp': '/tmp',
-
-  // If locales use different paths, you can
-  // specify each external path per locale.
-} satisfies Pathnames<typeof locales>;
-
-export const { Link, redirect, usePathname, useRouter, getPathname } =
-  createLocalizedPathnamesNavigation({ locales, localePrefix, pathnames });
+export const {Link, redirect, usePathname, useRouter} =
+  createSharedPathnamesNavigation({locales, localePrefix});
```
[the remark this answers is not in the document]
Comment (mohamedahmed-cloud) [reply navigation.ts:10]: https://next-intl-docs.vercel.app/docs/routing/navigation#localized-pathnames
look here

File: customHooks/Profile/useProfile.ts [hunk 3]
```diff
@@ -3,44 +3,37 @@
...
   return response.data;
 };
 
-export const useGetProfile = () => {
+export const useGetProfile = (enabled: boolean) => {
   return useQuery({
     queryKey: ['profile-get'],
     queryFn: profileGetFunction,
+    enabled: enabled ? enabled : true,
   });
```
[the remark this answers is not in the document]
Comment (mohamedahmed-cloud) [reply customHooks/Profile/useProfile.ts:26]: I will change it to `enabled: enabled `
باين كدة كنت نايم 😂

File: modules/Auth/Auth.ts
[the remark this answers is not in the document]
Comment (mohamedahmed-cloud) [reply modules/Auth/Auth.ts:1]: I use it to provide an isRegister function and use it anywhere in the component without the limitation of the hooks

**Pull request merged**